Bug/jenkins 46977 namespaces linkification

sophistifunk opened the pull request:
# Description

See [JENKINS-46977](https://issues.jenkins-ci.org/browse/JENKINS-46977).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

sophistifunk commented:
<img width="627" alt="screen shot 2017-09-20 at 1 01 20 pm" src="https://user-images.githubusercontent.com/142591/30628497-192409ae-9e1a-11e7-98e4-bff9db9c6dbe.png">

Old and busted

sophistifunk commented:
<img width="617" alt="screen shot 2017-09-20 at 1 14 57 pm" src="https://user-images.githubusercontent.com/142591/30628511-22b0bd64-9e1a-11e7-883a-64cbc6be6d66.png">

New hotness

michaelneale commented:
nice - if it's green, get it in. 

Is there a way to turn the story book into a unit test? as that is nice.

NicuPascu commented on jenkins-design-language/src/js/components/ResultItem.jsx:
```diff
@@ -9,6 +9,12 @@ import { getGlyphFor } from './status/SvgStatus';
 
+// Makes sure we only linkify explicit URLs with expected protocols
+function validateURL(value: string) {
+    const explicitURL = /^(http|https|file|ftp|mailto):/i;
```
add tel protocol maybe?

sophistifunk replied:
BO is not going to be nice on mobile devices any time soon, and we can easily add it if somebody asks, IMO. But if that's a general consensus it's easy to add.

michaelneale replied:
hrm tel could be interesting... click to call @bwalding and he will fix your build... hehehe

sophistifunk commented:
@michaelneale no nice way to turn stories into unit tests that I know of. There's a way to add in extra unit tests into your stories, but that seems dumb to me, as we're not running stories during build.

michaelneale commented:
@cliffmeyers ideas on how to unit test cover this ? (ATH is overkill for this).

sophistifunk commented:
We can write unit tests for it, it just doesn't seem worth the trouble to me because anybody messing with the component will be looking at the storybooks. I'll add a test tomorrow.

cliffmeyers commented:
@sophistifunk if you have a moment you should reach out to Thor. I know he's found some way to integrate storybooks and unit tests. I haven't looked close enough to decide whether I like it - kinda hard to wrap my head around how it would work - but if we avoid duplication of effort between storybook and unit test, that would be great.

michaelneale commented:
@sophistifunk if you update this, get it green, I will bee it as is I think

sophistifunk commented:
Added new unit tests for the linkify feature, and while I was in there I noticed the old tests weren't actually testing what they said they were testing, so I fixed them, then they flagged a bug, which is also fixed.

So, good call @michaelneale - but the process is still unbearable, thanks to our old-ass enzyme missing half the documented methods and jest eating all the logs.

michaelneale commented:
@sophistifunk ack.. yeah sounds painful. 🍺

when green - I think this should go in now. 

@sophistifunk need to make time to work all that stuff over, thanks for fixing things up.
